modules/channel-web/src/views/lite/components/messages/MessageGroup.tsx

```diff
@@ -7,69 +7,85 @@ import { Message as MessageDetails } from '../../typings'
-        'bpw-from-user': !props.isBot,
-        'bpw-from-bot': props.isBot
-      })}
-    >
-      {props.avatar}
-      <div className={'bpw-message-container'}>
-        {props.showUserName && <div className={'bpw-message-username'}>{props.userName}</div>}
-        <div className={'bpw-message-group'}>
-          {props.messages.map((data, i) => {
-            /**
-             * @deprecated 12.0
-             * Here, we convert old format to the new format Botpress uses internally.
-             * - payload: all the data (raw, whatever) that is necessary to display the element
-             * - type: extracted from payload for easy sorting
-             */
-            let payload = data.payload || data.message_data || data.message_raw || { text: data.message_text }
-            if (!payload.type) {
-              payload.type = data.message_type || (data.message_data && data.message_data.type) || 'text'
-            }
+class MessageGroup extends React.Component<Props> {
+  state = {
+    hasError: false
+  }
 
-            // Keeping compatibility with old schema for the quick reply
-            if (data.message_type === 'quick_reply' && !payload.text) {
-              payload.text = data.message_text
-            }
+  static getDerivedStateFromError(error) {
+    return { hasError: true }
+  }
 
-            if (data.message_type === 'file' && !payload.url) {
-              payload.url = (data.message_data && data.message_data.url) || (data.message_raw && data.message_raw.url)
-            }
+  render() {
+    if (this.state.hasError) {
+      return '* Cannot display message *'
+    }
 
-            if (props.messageWrapper && payload.type !== 'session_reset') {
-              payload = {
-                type: 'custom',
-                module: props.messageWrapper.module,
-                component: props.messageWrapper.component,
-                wrapped: payload
+    return (
+      <div
+        className={classnames('bpw-message-big-container', {
+          'bpw-from-user': !this.props.isBot,
+          'bpw-from-bot': this.props.isBot
+        })}
+      >
+        {this.props.avatar}
+        <div className={'bpw-message-container'}>
+          {this.props.showUserName && <div className={'bpw-message-username'}>{this.props.userName}</div>}
+          <div className={'bpw-message-group'}>
+            {this.props.messages.map((data, i) => {
+              /**
+               * @deprecated 12.0
+               * Here, we convert old format to the new format Botpress uses internally.
+               * - payload: all the data (raw, whatever) that is necessary to display the element
+               * - type: extracted from payload for easy sorting
+               */
+              let payload = data.payload || data.message_data || data.message_raw || { text: data.message_text }
+              if (!payload.type) {
+                payload.type = data.message_type || (data.message_data && data.message_data.type) || 'text'
```

Comment:
I like that you've documented the conversion but I'd extract this in a function for clarity.